tests/e2e/General/HTTPTest.php

```diff
@@ -131,7 +131,6 @@ public function testSpecs()
                 'content-type' => 'application/json',
             ], json_decode(file_get_contents($directory . $file), true));
 
-            $response['body'] = json_decode($response['body'], true);
```

Comment:
This removal is a bit concerning. What test is failing without it? Something doesn't seem right.

Reply:
It is not related to my changes, I just updated it to fix the failing test.

This is what the `body` looks like. 
```
["body"]=>
  array(1) {
    ["schemaValidationMessages"]=>
    array(0) {
    }
  }
  ```
  
  It is not a string. It is an array and the `json_decode` fails.

Reply:
Can you check if this test is passing fine in 1.6.x without removal? Would also be best to double check with Christy to find which PR started to fail this test, to be 100% sure we are not missing anything. Changes to general HTTP test are scarry.

Reply:
Yes, it is passing in 1.6.x before this PR. In this PR, we have added this line in `Client.php`: 
`$strpos = \is_bool($strpos) ? \strlen($responseType) : $strpos;`
If we don't add it, the test passes for `HttpTest.php`, however, I believe that code isn't correct because in case a `;` is not present, it doesn't consider the `responseType` as `application/json`. Would love for you to take a look here @christyjacob4 and @lohanidamodar (since you updated that test file last).

Reply:
Marking it as resolved as the logic was verified